testing/fil_token_integration/actors/basic_token_actor/src/lib.rs

```diff
@@ -47,23 +50,46 @@ impl FrcXXXToken<RuntimeError> for BasicToken<'_> {
         Ok(self.util.balance_of(&params)?)
     }
 
+    fn transfer(&mut self, params: TransferParams) -> Result<TransferReturn, RuntimeError> {
+        let spender = caller_address();
+        self.util.transfer(&spender, &spender, &params.to, &params.amount, &params.data)?;
+
+        // FIXME: populate with real values
+        Ok(TransferReturn { from_balance: TokenAmount::zero(), to_balance: TokenAmount::zero() })
```

Comment:
might be more efficient if Token::transfer returned the relevant info, else we can make separate calls to `Token::balance_of` and `Token::allowance` etc.

Reply:
Agreed. Simple but maybe error-prone would be to return an anonymous tuple. Another alternative is to return the Return types defined earlier, but that inverts the expected dependency hierarchy. Another option is separate return structs, but they'll be a bit duplicative.